src/components/ProductForm/ProductForm.tsx

```diff
@@ -1,59 +1,64 @@
 import { LoaderOverlay } from 'components/LoaderOverlay/LoaderOverlay';
-import { ReactElement } from 'react';
+import { ReactElement, useMemo } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPen } from '@fortawesome/free-solid-svg-icons';
-import { Dropdown } from 'components/Dropdown/Dropdown';
-import { useProductFormContext } from './ProductFormContext';
 import { Button } from 'components/Buttons/Button';
-import cn from 'classnames';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Select } from 'components/Inputs/Select';
+import { Input } from 'components/Inputs/Input';
+import { TextArea } from 'components/Inputs/TextArea';
+import { Checkbox } from 'components/Inputs/Checkbox';
+import { useProductFormContext } from './ProductFormContext';
+import { useCategories } from 'hooks/useCategories';
+import { useTags } from 'hooks/useTags';
+import { Label } from 'components/Inputs/Label';
+
+const formSchema = yup.object({
+  name: yup.string().required('Name is required').max(120, 'Name cannot be longer than 120 characters'),
+  desc: yup.string().optional().max(500, 'Description cannot be longer than 500 characters'),
+  price: yup.number().min(0, 'Price must be positive'),
+  category: yup.string().required('Category is required'),
+  stock: yup.number().required('Stock number is required').min(0).integer(),
+  tags: yup.array().of(yup.number()).required('Tag is required').min(1, 'At least one tag is required'),
+});
+
+type FormSchema = yup.InferType<typeof formSchema>;
 
 export const ProductForm = (): ReactElement => {
-  const formContext = useProductFormContext();
-  const [formProduct, setFormProduct] = formContext.useFormProduct;
-  const subHeadingClass = 'italic text-[#A2AAB6] font-poly';
+  const productForm = useProductFormContext();
+  const { tags } = useTags();
+  const { categories } = useCategories();
+  const {
```

Comment:
maybe this can be refactored alongside above too

Reply:
The manual setting of `defaultValues` here is necessary because of the mismatch of types between `ProductDto` and the schema, i.e. certain values in the form schema aren't in `ProductDto` like stock